app/src/main/java/chat/rocket/android/activity/MainActivity.java

```diff
@@ -91,25 +94,48 @@ private void setUserAwayIfServerAvailable() {
 
   private void setupSidebar() {
     SlidingPaneLayout pane = (SlidingPaneLayout) findViewById(R.id.sliding_pane);
-    if (pane != null) {
-      final SlidingPaneLayout subPane = (SlidingPaneLayout) findViewById(R.id.sub_sliding_pane);
-      pane.setPanelSlideListener(new SlidingPaneLayout.SimplePanelSlideListener() {
-        @Override
-        public void onPanelClosed(View panel) {
-          super.onPanelClosed(panel);
-          if (subPane != null) {
-            subPane.closePane();
-          }
-        }
-      });
+    if (pane == null) {
+      return;
+    }
 
-      Toolbar toolbar = (Toolbar) findViewById(R.id.activity_main_toolbar);
-      toolbar.setNavigationOnClickListener(view -> {
-        if (pane.isSlideable() && !pane.isOpen()) {
-          pane.openPane();
+    final SlidingPaneLayout subPane = (SlidingPaneLayout) findViewById(R.id.sub_sliding_pane);
+    pane.setPanelSlideListener(new SlidingPaneLayout.SimplePanelSlideListener() {
+      @Override
+      public void onPanelClosed(View panel) {
+        super.onPanelClosed(panel);
+        if (subPane != null) {
+          subPane.closePane();
         }
-      });
+      }
+    });
+
+    Toolbar toolbar = (Toolbar) findViewById(R.id.activity_main_toolbar);
+    toolbar.setNavigationOnClickListener(view -> {
+      if (pane.isSlideable() && !pane.isOpen()) {
+        pane.openPane();
+      }
+    });
+  }
+
+  private void setupToolbar() {
+    Toolbar toolbar = (Toolbar) findViewById(R.id.activity_main_toolbar);
+    setSupportActionBar(toolbar);
```

Comment:
I think it is better not to call `setSupportActionBar`. 
Because now we use very customized Toolbar, it is not good for the toolbar to be customized unexpectedly by Android framework.

Just put a customized Toolbar and setup it in each RoomFragment.